tests/ui/README.md

```diff
@@ -0,0 +1,123 @@
+
+``` javascript
+test('toggles state when clicked', () => {
+    const mockOnChange = jest.fn();
+    render(<Checkbox label="Accept Terms" onChange={mockOnChange} />);
+
+    const checkbox = screen.getByText('Accept Terms');
+    fireEvent.press(checkbox);
+    expect(mockOnChange).toHaveBeenCalledWith(true);
+
+    fireEvent.press(checkbox);
+    expect(mockOnChange).toHaveBeenCalledWith(false);
+});
+```
+### When to Skip UI Tests for Components:
+
+**Purely Presentational Components**:
+
+- Why: These components don’t contain logic or interactivity, so testing them is often unnecessary. Focus on visual output and accessibility instead.
+- Example: Avatar component that only displays an image, we generally don’t need tests unless there’s a specific behavior to verify.
+
+### Do’s
+
+- Write tests that reflect actual user behavior (e.g., clicking buttons, filling forms).
+- Mock external dependencies like network calls or Onyx data.
+- Use `findBy` and `waitFor` to handle asynchronous updates in the UI.
+- Follow naming conventions for test IDs (`testID="component-name-action"`).
+- Test for accessibility attributes like `accessibilityLabel`, `accessibilityHint`, etc., to improve test coverage for screen readers.
+- Reuse test utilities: Write helper functions to handle repetitive test setup, reducing code duplication and improving maintainability.
+- **Rather than targeting 100% coverage, prioritize meaningful tests for critical workflows and components**.
+
+### Don’ts
+
+- Don’t test implementation details (e.g., state variables or internal method calls).
+- Don’t ignore warnings: Fix `act()` or other common warnings to ensure reliability.
+- Avoid over-mocking: Mock dependencies sparingly to maintain realistic tests.
+    - *Bad*: Mocking a child component unnecessarily.
+    - *Good*: Mocking a network request while testing component interactions.
+- Don’t hardcode timeouts: Use dynamic queries like `waitFor` instead.
+
+###  When to Skip Tests
+
+Contributors may skip tests in the following cases:
+- Non-Functional Changes: E.g styling updates, github workflow changes, translation updates or refactors without behavioral changes.
+- Temporary Fixes: If a test will be added in a follow-up task, document the reason in the PR.
+
+**Note**: Always document skipped tests clearly in the PR description.
+
+### Common Pitfalls
+- Not awaiting async actions: Forgetting to await async actions can result in tests failing because components haven’t updated yet.
+
+```javascript
+// Correct usage
+await waitFor(() => expect(screen.getByText('Success')).toBeInTheDocument());
+```
+- Testing too much or too little: Striking a balance is key. Too many trivial tests lead to bloated test suites, while too few leave untested areas that could break easily.
+
+- Not cleaning up between tests: React components often leave behind side effects. Make sure to clean up between tests to ensure they are isolated and avoid conflicts.
+
```

Comment:
```suggestion
- Not cleaning up between tests: React components often leave behind side effects. Make sure to clean up between tests to ensure they are isolated and avoid conflicts.

- Having tightly coupled tests where one test depends on the actions from a previous test. This can cause tests to fail unexpectedly and it makes tests hard to maintain and debug. Each test should be self-sufficient.
```